### ISSUE TITLE
Issue merging files with anchors causing invalid YAML

### ISSUE THREAD

**Issue by dvianello** [mikefarah/yq#430]
**Describe the bug**
`yq` v3 seems to be producing invalid YAML when merging keys with archors

**Input Yaml**
data1.yml:
```yaml
vars:
  whitelist_tag: &whitelist_tag whitelist

policies:
  - filters:
      - type: value
        key: *whitelist_tag
```

data2.yml:
```yaml
vars:
  tagging_standard_present: &tagging_standard_present tag_present

policies:
  - filters:
      - and: *tagging_standard_present
```

**Command**
The command you ran:
```
yq m -a data1.yml data2.yml
```

**Actual behavior**

```yaml
vars:
  whitelist_tag: &whitelist_tag whitelist
  tagging_standard_present: tag_present
policies:
- filters:
  - type: value
    key: *whitelist_tag
- filters:
  - and: *tagging_standard_present
```

**Expected behavior**

```yaml
vars:
  whitelist_tag: &whitelist_tag whitelist
  tagging_standard_present: &tagging_standard_present tag_present
policies:
- filters:
  - type: value
    key: *whitelist_tag
- filters:
  - and: *tagging_standard_present
```

**Additional context**
YAML produced is invalid because it's missing an anchor `&tagging_standard_present `. Inverting the order files are specified for merging makes the other anchor (`&whitelist_tag`) disappear from the final output. 

Unrelated to this, we've also counted on `yq v2` exploding anchors on merge, which isn't happening anymore. This is causing downstream issues in our processing pipeline - any chance `--explodeAnchors` might be addetto to merge as well?

**Comment by mikefarah** [comment]
Sorry for the late response - I'm going to fix the alias merge issues.

For ```explodeAnchors``` (to keep things simple), I'd just pipe the merge output into another read with the explodeFlag.

**Comment by mikefarah** [comment]
Fixed in 3.3.4
